### DISCUSSION TITLE
turning off format-on-save?

### DISCUSSION THREAD

**Discussion by willpowellrtx** [interactive-data-language/vscode-idl#27] [Q&A] [answered]
Hi, I accidentally turned this on and now I'm unable to turn it off. Where is the setting stored?
I'm referring to:
"Ability to format your code on save and have control over key formatting preferences. The default matches what we think of as a modern approach and styling to IDL code that will be welcoming to new users."
Where are these settings stored/viewable/editable??

thanks
Will

**Comment by znorman-harris** [comment, marked as answer]
Hey Will!

The format on save setting is one that is managed by VSCode. The easiest way to change this is to open up your settings.json and delete the preferences we added in.

You can do this by:

1. Open the command palette in VSCOde (Control + Shift + p)
2. Type "user settings json" and open the entry that says: "Open User Settings (JSON)"
3. Remove these lines from the config:

```json
{
  "[idl]": {
    "editor.defaultFormatter": "idl.idl-for-vscode",
    "editor.formatOnSave": true
  }
}
```

4. Double check saving a file doesn't format automatically.

**Comment by SRockt1** [comment]
stern
